aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -327,16 +331,25 @@ std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool kee
 }
 
 std::tuple<Tensor, Tensor> max(const Tensor& self, int64_t dim, bool keepdim) {
-  Tensor max_indices = at::empty({0}, self.options().dtype(kLong));
-  if (self.is_quantized()) {
-    Tensor max = at::empty({0}, self.options().dtype(toUnderlying(self.scalar_type())));
-    at::native::max_out(self.int_repr() , dim, keepdim, max, max_indices);
-    // TODO: qscheme
-    return std::tuple<Tensor, Tensor>(at::_make_per_tensor_quantized_tensor(max, self.q_scale(), self.q_zero_point()), max_indices);
-  } else {
+  if (self.numel() == 0) {
     Tensor max = at::empty({0}, self.options());
+    Tensor max_indices = at::empty({0}, self.options().dtype(kLong));
+
     return at::native::max_out(self, dim, keepdim, max, max_indices);
   }
+  else {
```

Comment:
I don't understand what this diff is doing. When numel is zero you

```
    Tensor max = at::empty({0}, self.options());
    Tensor max_indices = at::empty({0}, self.options().dtype(kLong));

    return at::native::max_out(self, dim, keepdim, max, max_indices);
```

when numel is not zero and you're not quantized you do:

```
Tensor max_indices = at::empty({0}, self.options().dtype(kLong));
Tensor max = at::empty({0}, self.options());
return at::native::max_out(self, dim, keepdim, max, max_indices);
```

This looks exactly the same to me. The numel zero case doesn't handle quantization but this just seems like an oversight to me, as the special handling for quantized is required whether or not numel is zero or not.